src/controllers/profileInitialSetupController.js

```diff
@@ -163,13 +179,19 @@ const profileInitialSetupController = function (
       if (foundToken) {
         const expirationMoment = moment(foundToken.expiration);
 
-        if (expirationMoment.isAfter(currentMoment)) {
+        if (foundToken.isSetupCompleted) {
+          res.status(400).send(TOKEN_HAS_SETUP_MESSAGE);
+        } else if (foundToken.isCancelled) {
+          res.status(400).send(TOKEN_CANCEL_MESSAGE);
+        } else if (expirationMoment.isAfter(currentMoment)) {
```

Comment:
It's advisable to place the success condition as the final one.

Reply:
Updated.